pkg/remote/aws/repository/api_gateway_repository_test.go

```diff
@@ -0,0 +1,88 @@
+package repository
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/cloudskiff/driftctl/pkg/remote/cache"
+	awstest "github.com/cloudskiff/driftctl/test/aws"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/r3labs/diff/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_apigatewayRepository_ListAllRestApis(t *testing.T) {
+	tests := []struct {
+		name    string
+		mocks   func(client *awstest.MockFakeApiGateway)
+		want    []*apigateway.RestApi
+		wantErr error
+	}{
+		{
+			name: "list multiple rest apis",
+			mocks: func(client *awstest.MockFakeApiGateway) {
+				client.On("GetRestApisPages",
+					&apigateway.GetRestApisInput{},
+					mock.MatchedBy(func(callback func(res *apigateway.GetRestApisOutput, lastPage bool) bool) bool {
+						callback(&apigateway.GetRestApisOutput{
+							Items: []*apigateway.RestApi{
+								{Id: aws.String("restapi1")},
+								{Id: aws.String("restapi2")},
+								{Id: aws.String("restapi3")},
+							},
+						}, false)
+						callback(&apigateway.GetRestApisOutput{
+							Items: []*apigateway.RestApi{
+								{Id: aws.String("restapi4")},
+								{Id: aws.String("restapi5")},
+								{Id: aws.String("restapi6")},
+							},
+						}, true)
+						return true
+					})).Return(nil).Once()
+			},
+			want: []*apigateway.RestApi{
+				{Id: aws.String("restapi1")},
+				{Id: aws.String("restapi2")},
+				{Id: aws.String("restapi3")},
+				{Id: aws.String("restapi4")},
+				{Id: aws.String("restapi5")},
+				{Id: aws.String("restapi6")},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			store := cache.New(1)
+			client := awstest.MockFakeApiGateway{}
+			tt.mocks(&client)
+			r := &apigatewayRepository{
+				client: &client,
+				cache:  store,
+			}
+			got, err := r.ListAllRestApis()
+			assert.Equal(t, tt.wantErr, err)
+
+			if err == nil {
+				// Check that results were cached
+				cachedData, err := r.ListAllRestApis()
+				assert.NoError(t, err)
+				assert.Equal(t, got, cachedData)
+				assert.IsType(t, []*apigateway.RestApi{}, store.Get("apigatewayListAllRestApis"))
+			}
```

Comment:
You can use the cache mock to perform assertions instead. See snippet at https://github.com/cloudskiff/driftctl/blob/2fe76a506f844ed1086e9e3ffb4fbef0a8b0aa18/pkg/remote/google/repository/asset_test.go#L53-L55

We should update other tests as well someday

Reply:
Forgot about this one ! Good catch !